hw-4/data-access/group.repository.ts

```diff
@@ -0,0 +1,79 @@
+import { Group, UserGroup } from '@app-models/index';
+import uuid from 'uuid';
+import sequelize from '../sequelize/sequalize';
+
+
+export default class GroupRepository {
+  private readonly groupModel = Group;
+
+  private readonly userGroupModel = UserGroup;
+
+  getGroups(): Promise<Group[]> {
+    return this.groupModel.findAll();
+  }
+
+  getGroupById(id: string): Promise<Group> {
+    return this.groupModel.findOne({
+      where: {
+        id,
+      },
+    });
+  }
+
+  getGroupByName(name: string): Promise<Group|null> {
+    return this.groupModel.findOne({
+      where: {
+        name,
+      },
+    });
+  }
+
+  add(group: Group): Promise<Group> {
+    const { name, permissions } = group;
+
+    return this.groupModel.create<Group>({
+      id: uuid(),
+      name,
+      permissions,
+    }, {});
+  }
+
+  update(group: Group): Promise<Group> {
+    const {
+      id, name, permissions,
+    } = group;
```

Comment:
This is just cosmetics, though, we could either keep it all in a single line or break lines per property. Here I think the first is preferable. Also, we could use the destructuring directly on the `group` argument as we're not using the `group` for any other operation.

For ref: https://github.com/airbnb/javascript#destructuring--object
Note specially the quote: **Why? Destructuring saves you from creating temporary references for those properties.**

Reply:
Thanks for suggestion. We still need to use *id* in where statement. So I left it and replaced *update* body with destructuring.